### PR TITLE
fix panic caused by invalid multi-cluster secret

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
fixes: #12376

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
But we can not enforce how user provide the config

File: pkg/kube/secretcontroller/secretcontroller.go [hunk 1]
```diff
@@ -219,28 +221,38 @@ func (c *Controller) addMemberCluster(secretName string, s *corev1.Secret) {
...
 
 			clientConfig, err := LoadKubeConfig(kubeConfig)
 			if err != nil {
 				log.Infof("Data '%s' in the secret %s in namespace %s is not a kubeconfig: %v",
-					clusterID, secretName, s.ObjectMeta.Namespace, err)
+					clusterID, secretName, s.Namespace, err)
+				continue
+			}
+
+			if err := ValidateClientConfig(*clientConfig); err != nil {
```
Comment (mandarjog) [inline pkg/kube/secretcontroller/secretcontroller.go]: Is client config validated in Galley? seems like it should always be validated.

Comment (hzxuzhonghu) [reply pkg/kube/secretcontroller/secretcontroller.go]: Yes, I checked

Comment (costinm) [reply pkg/kube/secretcontroller/secretcontroller.go]: How could it be ? The secret should never be read by Galley ( shouldn't even have permissions )

Comment (hzxuzhonghu) [reply pkg/kube/secretcontroller/secretcontroller.go]: I misunderstanding the question, @costinm is right, the multi cluster secret is never read by galley

**Pull request merged**